functions/Get-DbaPageFileSetting.ps1

```diff
@@ -0,0 +1,136 @@
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program. If not, see http://www.gnu.org/licenses/.
+
+.LINK
+ https://dbatools.io/Get-DbaPageFileSetting
+
+.EXAMPLE
+Get-DbaPageFileSetting -ComputerName ServerA,ServerB
+
+Returns a custom object displaying ComputerName, AutoPageFile, FileName, Status, LastModified, LastAccessed, AllocatedBaseSize, InitialSize, MaximumSize, PeakUsage, CurrentUsage  for ServerA and ServerB
+
+.EXAMPLE
+ ServerA | Get-DbaPageFileSetting
+
+Returns a custom object displaying ComputerName, AutoPageFile, FileName, Status, LastModified, LastAccessed, AllocatedBaseSize, InitialSize, MaximumSize, PeakUsage, CurrentUsage  for ServerA
+
+#>[CmdletBinding()]
+	param (
+		[Parameter(Mandatory = $false, Position = 0, ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true)]
+		[Alias("cn", "host", "ServerInstance", "Server", "SqlServer")]
+		[object]$ComputerName,
+		[PSCredential] [System.Management.Automation.CredentialAttribute()]$Credential
+	)
+	BEGIN {}
+	PROCESS {
+        foreach ($Computer in $ComputerName)
+        {
+            Write-Verbose "Connecting to $Computer"
+			$reply = Resolve-DbaNetworkName -ComputerName $Computer -erroraction silentlycontinue
+            if ( $reply.ComputerName )
+            {
+                $computer = $reply.ComputerName
+            }
+            Write-Verbose "Connecting to $Computer via CIM (WSMan)"
+            $CompSys = Get-CimInstance -ComputerName $Computer -Query "SELECT * FROM win32_computersystem" -ErrorVariable $MyErr -Credential $Credential
+            if ( $CompSys )
+            {
+                Write-Verbose "Successfully retrieved PageFile information on $Computer via WSMan"
+                if ($CompSys.automaticmanagedpagefile -eq $False)
+                {
+                    $PF = Get-CimInstance -ComputerName $Computer -Query "SELECT * FROM win32_pagefile" # deprecated !
+                    $PFU = Get-CimInstance -ComputerName $Computer -Query "SELECT * FROM win32_pagefileUsage"
+                    $PFS = Get-CimInstance -ComputerName $Computer -Query "SELECT * FROM win32_pagefileSetting"
+                }
+            }
+			else
+            {
+				Write-Verbose "No WSMan connection to $Computer"
+                Write-Verbose "Getting computer information from server $Computer via CIM (DCOM)"
+			    $sessionoption = New-CimSessionOption -Protocol DCOM
+			    $CIMsession = New-CimSession -ComputerName $Computer -SessionOption $sessionoption -Credential $Credential

```

Comment:
I think this could be a little bit overwhelming. Imagine you have some CimSession in use for other module/command on the same session. That combo will remove all of them.
I would prefer to do that on the progress block after each one and remove only the one inside $CIMSession. What do you think?